runtime/litmus/src/tests/xcm_parachain/mod.rs

```diff
@@ -0,0 +1,260 @@
+	Origin,
+};
+use cumulus_primitives_core::{ParaId, PersistedValidationData};
+use cumulus_primitives_parachain_inherent::ParachainInherentData;
+
+use frame_support::{
+	assert_noop, assert_ok,
+	traits::{Currency, PalletInfoAccess},
+};
+use frame_system::RawOrigin;
+use polkadot_parachain::primitives::{AccountIdConversion, RelayChainBlockNumber, Sibling};
+use sp_runtime::{traits::Convert, AccountId32};
+use xcm::prelude::*;
+// use xcm::latest::prelude::*;
+use xcm_simulator::TestExt;
+
+pub mod relay_sproof_builder;
+
+fn para_a_account() -> AccountId32 {
+	ParaId::from(1).into_account()
+}
+
+fn para_b_account() -> AccountId32 {
+	ParaId::from(2).into_account()
+}
+
+fn _sibling_a_account() -> AccountId32 {
+	use sp_runtime::traits::AccountIdConversion;
+	Sibling::from(1).into_account()
+}
+
+fn _sibling_b_account() -> AccountId32 {
+	use sp_runtime::traits::AccountIdConversion;
+	Sibling::from(2).into_account()
+}
+
+#[test]
+fn test_xtokens_recognize_multilocation() {
+	TestNet::reset();
+	Relay::execute_with(|| {
+		let _ = pallet_balances::Pallet::<relay::Runtime>::deposit_creating(
+			&para_a_account(),
+			1_000_000_000_000,
+		);
+		let _ = pallet_balances::Pallet::<relay::Runtime>::deposit_creating(
+			&para_b_account(),
+			1_000_000_000_000,
+		);
+	});
+	ParaA::execute_with(|| {
+		register_channel_info(1, 2);
+	});
+	ParaB::execute_with(|| {
+		register_channel_info(2, 1);
+	});
+	ParaB::execute_with(|| {
+		assert_ok!(AssetManager::register_foreign_asset(
+			RawOrigin::Root.into(),
+			CurrencyId::ParachainReserve(Box::new(
+				(
+					Parent,
+					Parachain(1),
+					PalletInstance(<Balances as PalletInfoAccess>::index() as u8)
+				)
+					.into()
+			)),
+			Default::default()
+		));
+		assert_ok!(AssetManager::set_asset_units_per_second(
+			RawOrigin::Root.into(),
+			CurrencyId::ParachainReserve(Box::new(
+				(
+					Parent,
+					Parachain(1),
+					PalletInstance(<Balances as PalletInfoAccess>::index() as u8)
+				)
+					.into()
+			)),
+			1_000_000_000_000,
+			1
+		));
+	});
+
+	ParaA::execute_with(|| {
+		// Wrong Multilocation does not pass
+		assert_noop!(
+			XTokens::transfer(
+				Origin::signed(AccountId::from(ALICE)),
+				CurrencyId::SelfReserve,
+				1_000_000_000_000,
+				Box::new((Parent, Parachain(2)).into()),
+				800_000_000
+			),
+			orml_xtokens::Error::<Runtime>::NotSupportedMultiLocation
+		);
+
+		assert_ok!(XTokens::transfer(
+			Origin::signed(AccountId::from(ALICE)),
+			CurrencyId::SelfReserve,
+			1_000_000_000_000,
+			Box::new(
+				(Parent, Parachain(2), Junction::AccountId32 { network: Any, id: BOB }).into()
+			),
+			800_000_000
+		));
+	});
+
+	// TODO::This mimic does not finialized yet. Still need mimic XcmRouter
+	// ParaB::execute_with(|| {
+	// 	XcmpQueue::on_idle(0, u64::MAX);
+	// 	assert_eq!(Tokens::free_balance(0, &AccountId::from(BOB)), 999_200_000_000);
+	// })
+}
+
+#[test]
+fn test_pallet_xcm_recognize_multilocation() {
```

Comment:
Unfortunatley same problem 😂 

The test runs through even without `asset-manager` part. It's fine if it's meant to be a general integration test for polkadotXCM pallet, but not for `orml-xtokens/orml-tokens`

Reply:
same as above.  I put this as TODO already.